### PR TITLE
Updated APIs

### PR DESCRIPTION
**Pull request by Decoder07**
Fixes

- Updated Android and iOS API's

### CODE REVIEW

File: lib/src/hmssdk.dart [hunk 1]
```diff
@@ -377,15 +379,22 @@ class HMSSDK {
   /// [hmsActionResultListener] - the callback that would be called by SDK in case of a success or failure.
   void changeTrackStateForRole(
       {required bool mute,
-      required String type,
-      required String source,
-      required List<String> roles,
+      required HMSTrackKind? type,
```
Comment (ygit) [inline lib/src/hmssdk.dart:382]: `type` should be renamed to `kind`

File: lib/src/hmssdk.dart [hunk 2]
```diff
@@ -219,12 +219,16 @@ class HMSSDK {
...
-      required String roleName,
+      required List<HMSRole> roles,
       String? type,
       HMSActionResultListener? hmsActionResultListener}) async {
-    var arugments = {"message": message, "type": type, "role_name": roleName};
+
+    List<String> rolesMap = [];
+    roles.forEach((role)=> rolesMap.add(role.name));
+
+    var arugments = {"message": message, "type": type, "roles": rolesMap};
```
Comment (ygit) [inline lib/src/hmssdk.dart:229]: typo error: should be `arguments`
Also, `arguments` passed back to the listener would have HMSRole objects

**Review by ygit — CHANGES_REQUESTED** [review]

File: lib/src/hmssdk.dart [hunk 3]
```diff
@@ -377,13 +378,18 @@ class HMSSDK {
...
+
+    if (roles != null) roles.forEach((role) => rolesMap.add(role.name));
+
     var arguments = {
       "mute": mute,
-      "type": type,
+      "type": HMSTrackKindValue.getValueFromHMSTrackKind(
+          kind ?? HMSTrackKind.unknown),
       "source": source,
       "roles": roles
```
Comment (ygit) [inline lib/src/hmssdk.dart:394]: `rolesMap` should be sent here instead of `roles`?

File: lib/src/hmssdk.dart [hunk 4]
```diff
@@ -219,22 +219,25 @@ class HMSSDK {
...
       HMSActionResultListener? hmsActionResultListener}) async {
-    var arugments = {"message": message, "type": type, "role_name": roleName};
+    List<String> rolesMap = [];
+    roles.forEach((role) => rolesMap.add(role.name));
+
+    var arguments = {"message": message, "type": type, "roles": roles};
     var result = await PlatformService.invokeMethod(
         PlatformMethod.sendGroupMessage,
-        arguments: arugments);
+        arguments: arguments);
```
Comment (ygit) [inline lib/src/hmssdk.dart:231]: `rolesMap` should be sent here instead of `roles`?

Comment (ygit) [reply lib/src/hmssdk.dart:231]: To platform method, list of role name strings will be sent.
To the action result listener back to the app, list of HMSRole objects is to be sent

**Review by ygit — CHANGES_REQUESTED** [review]

File: lib/src/hmssdk.dart (hunk 4 above)
Comment (govindmaheshwari2) [reply lib/src/hmssdk.dart:231]: roles are list of HMSRole
rolesMap is list of String

**Review by ygit — APPROVED** [review]

**Pull request merged**